rfcs/table.md

```diff
@@ -0,0 +1,321 @@
+# Table Component
+
+Represents control to render tabular data.
+
+## Features
+- Sorting
+- Basic mobile responsiveness (overflow scrollbar)
+
+
+## Unsupported
+- Pagination
+- Filtering
+- Fixed Header
+- Fixed Column
+- Striped Tables
+- Hoverable row styling
+- Multi-column sort
+- Row headers
+- Expandable Row
+
+## Usage
+
+### Basic usage
+#### Render static tabular data.
+```javascript
+import * as React from 'react';
+import {Table} from 'baseui/table';
+const data = {
+    {
+        id: 1,
+        firstName: 'John'
+        lastName: 'Smith'
+    },
+    {
+            id: 1,
+            firstName: 'Lois'
+            lastName: 'Lane'
+    }
+};
+const columns = {
+    {
+        title: 'Employee ID'
+        dataIndex: 'id'
+    },
+    {
+        title: 'First'
+        dataIndex: 'firstName'
+    },
+    {
+        title: 'Last'
+        dataIndex: 'lastName'
+    },
+    {
+        title: 'Full Name'
+        render: (rowData) => {
+            const {firstName, lastName} = rowData;
+            return (
+                <span>
+                    `${firstName} ${lastName}`
+                </span>
+            );
+        }
+    }
+};
+export default () => <Table data={data} columns={columns}/>;
+```
+
+#### Styled Components
+```
+const CustomTd = styled(Td, props => ({
+  textColor: 'red',
+}));
+
+export default () => (
+ <Table
+    data={data}
+    columns={columns}
+    overrides={{ Td: CustomTd }}
+ />
+);
+```
+
+#### Sort
+
+```javascript
+const columns = [{
+  title: 'Name',
+  dataIndex: 'name',
+  sorter: (a, b) => a.name.length - b.name.length,
+}, {
+  title: 'Age',
+  dataIndex: 'age',
+  defaultSortOrder: 'descend',
+  sorter: (a, b) => a.age - b.age,
+}, {
+  title: 'Address',
+  dataIndex: 'address',
+  sorter: (a, b) => a.address.length - b.address.length,
+}];
+
+const data = [{
+  key: '1',
+  name: 'Sarah Brown',
+  age: 32,
+  address: 'New York No. 1 Anywhere',
+}, {
+  key: '2',
+  name: 'Jane Smith',
+  age: 42,
+  address: 'London No. 1 Anywhere',
+}, {
+  key: '3',
+  name: 'Joe Black',
+  age: 32,
+  address: 'Sidney No. 1 Anywhere',
+}, {
+  key: '4',
+  name: 'Jane Red',
+  age: 32,
+  address: 'London No. 2 Anywhere',
+}];
+
+export default = () => <Table columns={columns} dataSource={data}/>,
+```
+
+#### Custom Table Rendering
+There may be a case where more complex layouts are needed,
+such as columns that with colSpan > 1 and/or rowspan > 1
+```
+export default() => {
+    <Table
+     render: (dataSource, columns) => {
+        return (<Tbody>
```

Comment:
I'm not sure we actually want to render using HTML tables, given the conversation about using react-virtualized for out-of-the-box performance in this table component. Will defer to @CharlieMartell on this one.

Or is this just an example of completely custom rendering?

Reply:
Ohh, I see -- these are the names of the components you're exporting from the lib. K nevermind :)
Though I would be careful about using names so close to HTML element names, as consumers will expect the same functionality. If you intend to deviate from the HTML `<table>` spec, you might consider different (and more semantic) names, e.g. `<Header>`, `<Row>`, `<Cell>`.

Reply:
landed on semantic naming based on conversation today 👍